sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVector.java

```diff
@@ -143,7 +150,27 @@ public ColumnarArray getArray(int rowId) {
 
   @Override
   public ColumnarMap getMap(int ordinal) {
-    throw new IllegalStateException("Maps are currently not supported by rapids cudf");
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> structHcv = cudfCv.getChildColumnViewAccess(0);
+    // keys
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> firstHcv = structHcv.getChildColumnViewAccess(0);
+    HostColumnVectorCore firstHcvCore = new HostColumnVectorCore(firstHcv.getDataType(),
+        firstHcv.getRowCount(), Optional.of(firstHcv.getNullCount()), firstHcv.getDataBuffer(),
+        firstHcv.getValidityBuffer(), firstHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+    // values
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> secondHcv = structHcv.getChildColumnViewAccess(1);
+    HostColumnVectorCore secondHcvCore = new HostColumnVectorCore(secondHcv.getDataType(),
+        secondHcv.getRowCount(), Optional.of(secondHcv.getNullCount()), secondHcv.getDataBuffer(),
+        secondHcv.getValidityBuffer(), secondHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+
+    RapidsHostColumnVectorCore firstChild = new RapidsHostColumnVectorCore(
+        GpuColumnVector.getSparkType(firstHcvCore.getType()), firstHcvCore);
+    RapidsHostColumnVectorCore secondChild = new RapidsHostColumnVectorCore(
+        GpuColumnVector.getSparkType(secondHcvCore.getType()), secondHcvCore);
+    //TODO: test more that offset and len are right
+    return new ColumnarMap(firstChild, secondChild,
+        ordinal * DType.INT32.getSizeInBytes(), (ordinal + 1) * DType.INT32.getSizeInBytes());
```

Comment:
The values to to offset and length are wrong.  They are indexes into first child and second child, not byte offsets into memory.  You need to get the start offset and end offset from cudfCv.

Reply:
Yea I added the TODO here so that i can get some comments on that. Thank u for pointing out what it should be. To me the ColumnarArray offset in spark code threw me off.

Reply:
Updated for further comments